lib/ffmpeg.rb

```diff
@@ -110,6 +111,27 @@ def ffmpeg_binary
       @ffmpeg_binary ||= which('ffmpeg')
     end
 
+    # Get the version of the ffmpeg binary.
+    #
+    # @return [String] The version string (e.g., "4.4.6", "8.0")
+    def ffmpeg_version
+      @ffmpeg_version ||= begin
+        stdout, = FFMPEG::IO.capture3(ffmpeg_binary, '-version')
+        stdout[/ffmpeg version (\d+\.\d+(?:\.\d+)?)/i, 1]
+      end
+    end
+
+    # Check if the ffmpeg version matches the given pattern.
+    #
+    # @param pattern [String, Regexp] The version pattern to match.
+    # @return [Boolean] True if the ffmpeg version matches the pattern, false otherwise.
+    def ffmpeg_version?(pattern)
+      return false unless ffmpeg_version
+      return !!(ffmpeg_version =~ pattern) if pattern.is_a?(Regexp)
```

Comment:
the `=~` method also updates some global variables that we're not using. if the pattern is a Regexp, I would just use `match?` that will return only the required boolean

Reply:
Fixed

Reply:
you left it as `pattern.match?(pattern)`, one of them should be ffmpeg_version